src/Workspaces/Core/Portable/Workspace/Solution/Project.cs

```diff
@@ -227,19 +226,40 @@ public bool ContainsAnalyzerConfigDocument(DocumentId documentId)
         /// Get the document in this project with the specified document Id.
         /// </summary>
         public Document? GetDocument(DocumentId documentId)
-            => ImmutableHashMapExtensions.GetOrAdd(ref _idToDocumentMap, documentId, s_tryCreateDocumentFunction, this);
+        {
+            if (ContainsDocument(documentId))
```

Comment:
ImmutableHashMapExtensions.GetORAdd had an important behavior that meant if the lookup into the dictionary missed, it'd call the creation function, and if it returned null would then return null. The standard ImmutableInterlocked.GetOrAdd does not do such a thing; this pattern is the pattern we use in Solution.GetProject so I'm using it here until we have data to argue for something better/different.